naga/REVIEW-CHECKLIST.md

```diff
@@ -0,0 +1,55 @@
+# Naga Review Checklist
+
+This is a collection of notes on things to watch out for when
+reviewing pull requests submitted to Naga.
+
+Unfortunately, there are a few parts of Naga where certain knowledge,
+requirements, or concerns are spread out across several areas of the
+code. This makes mistakes easier, as it is more likely for a
+contributor to fix one spot, but neglect others.
+
+Ideally, one fixes this kind of problem by refactoring the code so
+that requirements are easier to identify with local reasoning. We do
+this regularly. But when such projects are too large to undertake
+immediately, we can use review checklists as a temporary workaround
+to ensure that all the different spots do get addressed.
+
+Some of these points are also just coding style issues. Such problems
+are probably better to address with comments in the code, or effective
+use of Rust (exhaustive match checking, for example).
+
+## General
+
+If your change iterates over a collection, did you ensure the order of
+iteration was deterministic? Using `HashMap` and `HashSet` is fine, as
+long as you don't iterate over it.
+
+## IR changes
+
+If your change adds or removes `Handle`s from the IR:
```

Comment:
style: Prettier would add a newline between this paragraph and the list following it. Might be nice for avoiding churn, but no blocking here.

Reply:
This also applies to a couple of other places here.

Reply:
That's surprising to me, because whether or not there's a newline before an itemized list has semantic significance in CommonMark, at least: it indicates whether a list is "tight" or "loose". [spec](https://spec.commonmark.org/0.31.2/#loose). I would like these lists to be tight.